subsys/net/lib/lwm2m/lwm2m_message_handling.c

```diff
@@ -175,8 +175,10 @@ static int get_block_ctx(const struct lwm2m_obj_path *path, struct lwm2m_block_c
 	*ctx = NULL;
 
 	for (i = 0; i < NUM_BLOCK1_CONTEXT; i++) {
-		if (memcmp(path, &block1_contexts[i].path,
-			  sizeof(struct lwm2m_obj_path)) == 0) {
+		if (path->level == block1_contexts[i].path.level &&
+		    path->obj_id == block1_contexts[i].path.obj_id &&
+		    (path->level < 2 || path->obj_inst_id == block1_contexts[i].path.obj_inst_id) &&
```

Comment:
nit: Could use dedicated symbols (`LWM2M_PATH_LEVEL_OBJECT_INST`, `LWM2M_PATH_LEVEL_RESOURCE`) instad of magic numbers

Reply:
Is this comparison leading to same result as using `lwm2m_obj_path_equal(path, block_contextx.path)`

See lwm2m_util.c